src/device_info.sh

```diff
@@ -0,0 +1,413 @@
+
+  cpu_currently=${cpu_currently//[!0-9,.]/}
+  cpu_max=${cpu_max//[!0-9,.]/}
+  cpu_min=${cpu_min//[!0-9,.]/}
+
+  device_info_data['cpu_currently']="$cpu_currently"
+  device_info_data['cpu_max']="$cpu_max"
+  device_info_data['cpu_min']="$cpu_min"
+}
+
+# This function populates the values from the size and fs (filesystem) key of
+# the device_info_data variable.
+#
+# @target Target machine
+function get_disk()
+{
+  local target="$1"
+  local flag="$2"
+  local ip="$3"
+  local port="$4"
+  local cmd
+  local info
+  local size
+  local mount
+  local fs
+
+  target=${target:-"${device_options['target']}"}
+  ip=${ip:-"${device_options['ip']}"}
+  port=${port:-"${device_options['port']}"}
+  flag=${flag:-'SILENT'}
+  cmd="df -h / | tail -n 1 | tr -s ' '"
+  case "$target" in
+    1) # VM_TARGET
+      cmd="df -h ${configurations[mount_point]} | tail -n 1 | tr -s ' '"
+      info=$(cmd_manager "$flag" "$cmd")
+      ;;
+    2) # LOCAL_TARGET
+      info=$(cmd_manager "$flag" "$cmd")
+      ;;
+    3) # REMOTE_TARGET
+      info=$(cmd_remotely "$cmd" "$flag" "$ip" "$port")
+      ;;
+  esac
+
+  if [[ "$flag" == 'TEST_MODE' ]]; then
+    echo "$info"
+    return 0
+  fi
+
+  fs="$(echo "$info" | cut -d' ' -f1)"
+  size="$(echo "$info" | cut -d' ' -f2)"
+  mount="$(echo "$info" | cut -d' ' -f6)"
+
+  device_info_data['disk_size']="$size"
+  device_info_data['root_path']="$fs"
+  device_info_data['fs_mount']="$mount"
+}
+
+# This function populates the os variable from the device_info_data variable.
+#
+# @target Target machine
+function get_os()
+{
+  local target="$1"
+  local ip="$2"
+  local port="$3"
+
+  local os
+
+  target=${target:-"${device_options['target']}"}
+  ip=${ip:-"${device_options['ip']}"}
+  port=${port:-"${device_options['port']}"}
+  case "$target" in
+    1) # VM_TARGET
+      os=$(detect_distro "${configurations[mount_point]}")
+      ;;
+    2) # LOCAL_TARGET
+      os=$(detect_distro '/')
+      ;;
+    3) # REMOTE_TARGET
+      os=$(which_distro "$ip" "$port")
+      ;;
+  esac
+
+  device_info_data['os']="$os"
+}
+
+# This function populates the gpu associative array with the vendor and
+# fetchable memory from each GPU found in the target machine.
+function get_gpu()
+{
+  local target="$1"
+  local flag="$2"
+  local ip="$3"
+  local port="$4"
+  local cmd
+  local pci_addresses
+  local gpu_info
+
+  target=${target:-"${device_options['target']}"}
+  ip=${ip:-"${device_options['ip']}"}
+  port=${port:-"${device_options['port']}"}
+  flag=${flag:-'SILENT'}
+  cmd="lspci | grep -e VGA -e 3D | cut -d' ' -f1"
+  case "$target" in
+    2) # LOCAL_TARGET
+      pci_addresses=$(cmd_manager "$flag" "$cmd")
+      for g in $pci_addresses; do
+        gpu_info=$(cmd_manager "$flag" "lspci -v -s $g")
+        gpu_name=$(echo "$gpu_info" | sed -nr '/Subsystem/s/\s*.*:\s+(.*)/\1/p')
+        gpu_provider=$(echo "$gpu_info" | sed -nr '/controller/s/.+controller: *([^ ]+).+/\1/p')
```

Comment:
You need to update your regex, in my case, I'm your function return `Advanced instead of ` Advanced Micro Devices`.